File: data/samples/sample_code.py
import os
import sys
import os  # duplicate import

def calculate(x,y,operation):
    if operation == "add":
        return x+y
    elif operation == "subtract":
        return x-y
    elif operation == "multiply":
        return x*y
    elif operation == "divide":
        if y == 0:
            return None  # should raise an exception
        return x/y
    elif operation=="power":
        return x**y
    else:
        return None  # silent failure

def process_data(data):
    result = []
    for i in range(len(data)):   # should use enumerate
        if data[i] != None:      # should use `is not None`
            result.append(data[i] * 2)
    return result

password = "supersecret123"   # hardcoded secret!

def unused_function():
    pass

File: data/samples/sample_code.js
var password = "supersecret123"  // hardcoded secret

var x = 10
var y = 0

function calculate(x, y, operation) {
    if (operation == "add") return x + y
    if (operation == "subtract") return x - y
    if (operation == "multiply") return x * y
    if (operation == "divide") return x / y  // no zero check!
}

function processData(data) {
    var result = []
    for (var i = 0; i < data.length; i++) {
        if (data[i] != null) {   // should use !==
            result.push(data[i] * 2)
        }
    }
    return result
}

var unused = "this variable is never used"

console.log(calculate(x, y, "divide"))  // will return Infinity